src/model/wallet.ts

```diff
@@ -257,11 +258,17 @@ export const walletInit = async (
   return { isNew, walletAddress };
 };
 
-export const loadWallet = async (
-  address?: EthereumAddress | undefined,
+export const loadWallet = async <S extends Screen>({
+  address,
   showErrorIfNotLoaded = true,
-  provider?: Provider
-): Promise<null | Wallet | LedgerSigner> => {
+  provider = web3Provider,
```

Comment:
I know this is likely a noop but let's leave these default changes out of this PR for now to reduce scope for testing for QA, and any improvements we want to make we can make a separate PR for (e.g, keeping `provider` as optional and not defaulting it to the `web3Provider`)

Reply:
It already defaults to web3Provider if you don't pass it later in this code block. This was mostly just to clean up typescript types.

Reply:
yeah understood, it's likely a noop but it just technically introduces a potential path change difference in the event of a hardware wallet call that isn't passing the `provider` (on line 298 , which shouldn't happen) - basically what I'm getting at directionally is: since this is a sensitive PR in terms of touching core txn flows and the wallet model, let's reduce all variables as much as possible except for the changes strictly needed, and we can come back and improve the rest in a separate test

Reply:
👍🏼 reverted